django/db/models/sql/subqueries.py

```diff
@@ -142,7 +142,13 @@ def add_update_fields(self, values_seq):
         that will be used to generate the UPDATE query. Might be more usefully
         called add_update_targets() to hint at the extra information here.
         """
-        self.values.extend(values_seq)
+        for field, model, val in values_seq:
+            if hasattr(val, 'resolve_expression'):
+                # Resolve expressions here so that annotations are no longer needed
+                val = val.resolve_expression(self, allow_joins=False, for_save=True)
+                if val.contains_aggregate:
+                    raise FieldError("Aggregate functions are not allowed in this query")
```

Comment:
This new code path is not reached by any of your addition tests. Is it supposed to triggered